templates/install_puppet.bat.erb

```diff
@@ -1,22 +1,36 @@
+SET
+
 set AGENT_PID=%1
 set windowTitle=Puppet Agent Upgrade
 title %windowTitle%
 
 set pid=
 for /f "tokens=2" %%a in ('tasklist /v ^| findstr /c:"%windowTitle%"') do set pid=%%a
-set pid_path=%TEMP%\puppet_agent_upgrade.pid
+set pid_path=%~dp0puppet_agent_upgrade.pid
```

Comment:
Hmm... and we're certain this is always writable?  And that there are no side effects of writing into this Puppet module template dir (consider PMT install vs pluginsync)?

I think it would be preferable to just set `%TEMP%` to `%windir%\Temp` if it's not already set:

```batch
IF "%TEMP%"=="" (SET TEMP=%windir%\Temp)
```